src/index.js

```diff
@@ -79,7 +79,16 @@ const {
 
 /**
  * @template T
- * @typedef {InferDefaultType<T> | undefined} BasicTransformerOptions
+ * @typedef {Object} BasicTransformerOptions
+ * @property {ResizeOptions} [resize]
```

Comment:
I am fine with it, but in future we can add more options for the query syntax and use there not only width/height, so make sense to use better name here

Reply:
I reverted `BasicTransformerOptions` type definition for now. I think that it should look like this:
```js
/**
 * @template T
 * @typedef {Object} BasicTransformerOptions
 * @property {ResizeOptions} [resize]
 * @property {RotateOptions} [rotate]
 * @property {SomeOtherTransformOptions} [.......]
 * @property {InferDefaultType<T> | undefined} [encodeOptions]
```
but not sure that `InferDefaultType` for `encodeOptions` does not braiking types copabylity with previous versions. And these changes also requires normalize `RotateOptions` between implementations (`squoosh` and `sharp`), otherwise they be inconsistent.